.github/workflows/main.yml

```diff
@@ -0,0 +1,104 @@
+name: CI
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  build:
+    name: "Build & Test"
+    runs-on: ${{ matrix.config.os }}
+    strategy:
+      matrix:
+        config:
```

Comment:
I wonder: do we need to test all of the Ubuntu+compiler combinations?

We could focus on one version of Ubuntu, and "all the compilers" on that Ubuntu?